[ACIX-1010] Migrate go module creation how-to docs

Ishirui opened the pull request:
### What does this PR do?

Moves the `docs/dev/modules.md` file to the new docs site, merging it with the existing `docs/dev/public/how-to/go/add-module.md`.

### Motivation

Streamlining, updating and generally improving our docs.

### Describe how you validated your changes

Running `dda docs serve` and checking the website looks as expected + manually checking for any broken links.

### Additional Notes

Ishirui committed:
refactor(docs): Merge `add_module.md` and first section of old `docs/dev/modules.md`

Ishirui committed:
refactor(docs): Finish migrating `docs/dev/modules.md`

Ishirui committed:
fix(docs): Add new file to nav

ofek commented on docs/public/how-to/go/modules.md:
```diff
@@ -0,0 +1,168 @@
+## Creating a new module
+
+1. Determine the packages that you want to expose and their dependencies.
+/// info
```
Everything under each list item should be indented.

Ishirui committed:
refactor(docs): Indent under list items

For this to work properly with some formatting elements (admonitions and `>` blocks), we also need to add an extra newline between the list item and the indented section.

Ishirui commented:
/merge
